### DISCUSSION TITLE
I swapped SQLx for Cornucopia

### DISCUSSION THREAD

**Discussion by ianpurton** [cornucopia-rs/cornucopia#26] [General]
Just to let you know I'm now running Cornucopia in https://cloak.software/

The code is here https://github.com/purton-tech/cloak

In particular the queries and build.rs are in the following folder https://github.com/purton-tech/cloak/tree/main/app

I'm pleased with the results, I completely removed SQLx and was able to get faster build times as well as reduction in project dependencies.

I like the way the queries are separated out into their own files.

I love the format of the .sql files, very concise.

For the function calls I noticed we always pass in a reference even for i32 etc. So I get code like the following.

```
queries::service_accounts::delete_service_account(
        &client,
        &service_account_id,
        &user_id),
   )
```

Not a big deal but thought I would mention it.

Thanks.

**Comment by LouisGariepy** [comment]
Awesome, I'm really impressed with your work on cloak! I took a glance at the source, good stuff. (As a sidenote, it looks like you forgot to remove the `sqlx-data.json` files that `sqlx` generates in offline mode). I also really liked you `Rust on Nails` article :smile:.

Eventually, I'd like to add a `Awesome Cornucopia` section to the readme, linking to impressive projects using cornucopia, and cloak would be a prime candidate (you willing, of course).

Regarding passing by reference, `tokio_postgres` expects SQL parameter values to be passed as references (even small ` Copy` types like `i32`), and, thus, so must cornucopia. The references could probably be taken under the hood for types that are known to be `Copy`, but I figured the consistency of always passing by reference would be less confusing.

Thanks for kind words, I'm honestly stoked that Cornucopia is useful to the community. Feel free to report new issues and feature suggestions as you come by them, even small things. I'm always looking for ways to improve the crate.
